src/goto-programs/goto_inline_class.cpp

```diff
@@ -338,7 +339,7 @@ void goto_inlinet::insert_function_body(
   inline_log.copy_from(goto_function.body, body);
 
   goto_programt::instructiont &end=body.instructions.back();
-  assert(end.is_end_function());
+  DATA_INVARIANT(end.is_end_function(), "end of function");
```

Comment:
If possible could the message include a short explanation of why this should be impossible? Perahs `"final instruction of a function must be an END_FUNCTION"`